### ISSUE TITLE
loading data: dataset.id needs to be specified

### ISSUE THREAD

**Issue by brochhagen** [clics/clics2#50]
While following the README instructions I run into an issue when loading (all available) data.

When running
`> clics load ../concepticon-data/ ../glottolog/`

with the README-specified version of glottolog/concepticon; and a clean python 3.7 conda environment, I get

`ValueError: Dataset.id needs to be specified in subclass for <class 'lexibank_allenbai.Dataset'>!
`

This seems like a mismatch between the attributes of the dataset(s) and what CLICS expects. I tried to navigate around the problem, but I ended up not achieving much.

I know this is vague, but I'd appreciate any pointers to solve this problem. Thanks!

 Here's the full traceback

 ```INFO    using IClicsForm implementation pyclics.plugin:clics_form
INFO    loading datasets into clics.sqlite
Traceback (most recent call last):
  File "/home/u167750/miniconda3/envs/clics/bin/clics", line 10, in <module>
    sys.exit(main())
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/pyclics/__main__.py", line 33, in main
    sys.exit(parser.main(parsed_args=args))
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/clldutils/clilib.py", line 110, in main
    catch_all=catch_all, parsed_args=args)
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/clldutils/clilib.py", line 82, in main
    self.commands[args.command](args)
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/clldutils/clilib.py", line 35, in __call__
    return self.func(args)
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/pyclics/commands.py", line 108, in load
    for ds in iter_datasets():
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/pylexibank/dataset.py", line 697, in iter_datasets
    yield ep.load()(glottolog=glottolog, concepticon=concepticon)
  File "/home/u167750/miniconda3/envs/clics/lib/python3.7/site-packages/pylexibank/dataset.py", line 206, in __init__
    "Dataset.id needs to be specified in subclass for %s!" % self.__class__)
ValueError: Dataset.id needs to be specified in subclass for <class 'lexibank_allenbai.Dataset'>!
```

**Comment by LinguList** [comment]
You have version clashes, I guess. Did you make sure to follow our instructions completely, especially installing versions from datasts.txt?

**Comment by LinguList** [comment]
The reason is: we're working on clics3 now, and this will build on quite a few new versions of different software parts, which will be backwards-incompatible with clics2.

**Comment by chrzyki** [comment]
Heya,

the following steps should help with this:

- create a new virtual environment
- `pip install -r requirements.txt` with the CLICS2 requirements.txt
- `pip install -r datasets.txt` with the CLICS2 datasets.txt
- `git checkout 9701cb0` for Glottolog, `git checkout tags/v1.2.0` for Concepticon
- clics load path/to/concepticon-data path/to/glottolog

I'm assuming (just assuming, sorry if that's not the case) that you're using a newer version of `pylexibank`, which unfortunately is not compatible anymore with the older releases. The `pylexibank==0.9.0` in the CLICS2 requirements.txt should help with that.

**Comment by chrzyki** [comment]
Just tested it - the above should produce a `clics.sqlite` with stats identical to the ones specified in the README (i.e. 2487 Concepts, 1220 Varieties, 1028 Glottocodes, 90 Families). Let me know if that is not the case or if you need any additional help. :)

**Comment by LinguList** [comment]
Thanks, Christoph, @brochhagen, please confirm if this all works, so we can close this issue.

**Comment by brochhagen** [comment]
Thanks a lot, @chrzyki and @LinguList! The problem was indeed solved by using the requirements.txt file in my fresh virtual environment. I somehow missed that very crucial step. Sorry for the inconvenience.

Some tangential notes on installing from the requirements: I ran into two of issues that are not related to CLICS per se. However, maybe someone else will have the same problems and will find this useful:
  * Github and SSH: I had a `Permission denied (publickey). fatal: Could not read from remote repository`-error for the `pyclics` installation. Easily fixed by changing line 33 in requirements.txt to `-e git+https://github.com/clics/clics2.git@97b3121bc2f94c7f417184dc1ffdb8b276a809f8#egg=pyclics
`
  * pip and global env: I got `Cannot uninstall 'certifi'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.` that was preventing the installation of `python-igraph`. Some explanations and workarounds are discussed extensively, e.g., [here](https://github.com/pypa/pip/issues/5247). In my case temporarily downgrading to pip 9.0.3, as described [here](https://github.com/pypa/pip/issues/5247#issuecomment-410910018), worked.

 should someone have similar issues:
